### DISCUSSION TITLE
Cannot find module gatsby-page-utils or/and gatsby-core-utils

### DISCUSSION THREAD

**Discussion by krzy789** [gatsbyjs/gatsby#38074] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Running tests by jest is causing this kind of errors.

on my repo 

>     Cannot find module 'gatsby-core-utils/create-content-digest' from 'node_modules/gatsby/cache-dir/commonjs/slice/server-        slice.js'
> 
>     Require stack:
>       node_modules/gatsby/cache-dir/commonjs/slice/server-slice.js
>       node_modules/gatsby/cache-dir/commonjs/slice.js
>       node_modules/gatsby/cache-dir/commonjs/gatsby-browser-entry.js

on reproduction

> Cannot find module 'gatsby-page-utils/apply-trailing-slash-option' from 'node_modules/gatsby-link/dist/index.js'
> 
>     Require stack:
>       node_modules/gatsby-link/dist/index.js
>       node_modules/gatsby/cache-dir/commonjs/gatsby-browser-entry.js

Here is the source of the problem https://github.com/krzy789/gatsby-navigate-error/blob/main/src/containers/reducers.js#L2

### Reproduction Link

https://github.com/krzy789/gatsby-navigate-error

### Steps to Reproduce

1. npm install
2. npm run test

### Expected Result

Tests should pass/break without this kind of errors.

### Actual Result

Cannot find module 'gatsby-page-utils/apply-trailing-slash-option' from 'node_modules/gatsby-link/dist/index.js
Cannot find module 'gatsby-page-utils/apply-trailing-slash-option' from 'node_modules/gatsby-link/dist/index.js

### Environment

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (12) x64 Intel(R) Core(TM) i7-10850H CPU @ 2.70GHz
  Binaries:
    Node: 18.16.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.17 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 8.12.1 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1555.0), Chromium (113.0.1774.35)
  npmPackages:
    gatsby: ^5.9.0 => 5.9.0
```

**Comment by LekoArts** [comment]
Use Jest 28 or 29 and it'll work.

For Jest 27 you have to configure the `moduleNameMapper` as shown here: https://github.com/gatsbyjs/gatsby/blob/24c393e1218b025fb64ea2a66f7bbe8eb1471366/docs/docs/how-to/testing/unit-testing.md
